kornia/losses/perceptual.py

```diff
@@ -0,0 +1,304 @@
+        self.resnet.avgpool = torch.nn.Identity()
+        self.resnet.fc = torch.nn.Identity()
+
+        # Freeze the model
+        if freeze:
+            for param in self.resnet.parameters():
+                param.requires_grad = False
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+
+        if x.shape[1] == 1:
+            x = x.repeat(1, 3, 1, 1)
+
+        x = self.resnet.conv1(x)
+        x = self.resnet.bn1(x)
+        x = self.resnet.relu(x)
+        x = self.resnet.maxpool(x)
+
+        x = self.resnet.layer1(x)
+
+        if self.output_layer > 1:
+            x = self.resnet.layer2(x)
+        if self.output_layer > 2:
+            x = self.resnet.layer3(x)
+        if self.output_layer > 3:
+            x = self.resnet.layer4(x)
+
+        return x
+
+
+def _image_shape_to_corners(image: torch.Tensor) -> torch.Tensor:
+    """Convert image size to 4 corners representation in clockwise order.
+
+    Args:
+        image: image tensor with shape :math:`(B, C, H, W)` where B = batch size,
+            C = number of channels
+
+    Return:
+        the corners of the image.
+    """
+    if len(image.shape) != 4:
+        raise ValueError('patch should be of size B, C, H, W')
+    batch_size = image.shape[0]
+    image_width = image.shape[-2]
+    image_height = image.shape[-1]
+    corners = torch.tensor(
+        [[0, 0], [image_width, 0], [image_width, image_height], [0, image_height]],
+        device=image.device,
+        dtype=image.dtype,
+        requires_grad=False,
+    )
+    corners = corners.repeat(batch_size, 1, 1)
+
+    return corners
+
+
+def _four_point_to_homography(corners: torch.Tensor, deltas: torch.Tensor) -> torch.Tensor:
+    """Convert 4-point representation introduced in :cite:`detone2016deep` to homography.
+
+    Args:
+        corners: corners tensor with shape :math:`(B, 4, 2)` where B = batch size
+        deltas: deltas tensor with shape :math:`(B, 4, 2)` where B = batch size
+
+    Return:
+        the converted homography.
+    """
+
+    if not isinstance(corners, torch.Tensor):
+        raise TypeError(f"corners type is not a torch.Tensor. Got {type(corners)}")
+
+    if not isinstance(deltas, torch.Tensor):
+        raise TypeError(f"deltas type is not a torch.Tensor. Got {type(deltas)}")
+
+    if not len(corners.shape) == 3 or not corners.shape[1] == 4 or not corners.shape[2] == 2:
+        raise ValueError(f"Invalid input shape of corners, we expect Bx4x2. Got: {corners.shape}")
+
+    if not len(deltas.shape) == 3 or not deltas.shape[1] == 4 or not deltas.shape[2] == 2:
+        raise ValueError(f"Invalid input shape of deltas, we expect Bx4x2. Got: {deltas.shape}")
+
+    if not corners.size(0) == deltas.size(0):
+        raise ValueError(f'Expected corners batch_size ({corners.size(0)}) to match deltas batch '
+                         f'size ({deltas.size(0)}).')
+
+    corners_hat = corners + deltas
+    homography_inv = get_perspective_transform(corners, corners_hat)
```

Comment:
```suggestion
    dst_homo_src = get_perspective_transform(corners, corners_hat)
```
i'd suggest to follow our internal convention to describe linear transformations
http://paulfurgale.info/news/2014/6/9/representing-robot-pose-the-good-the-bad-and-the-ugly